doc/lsst.pipe.base/creating-a-command-line-task.rst

```diff
@@ -0,0 +1,253 @@
+
+- They are subclasses of `lsst.pipe.base.CmdLineTask`, whereas regular tasks are subclasses of `lsst.pipe.base.Task`.
+- They have an associated :ref:`run script <creating-a-command-line-task-run-script>` to run them from the command-line as pipelines (this is common, but not required, for regular tasks).
+- They have a ``run`` method which performs the full pipeline data processing.
+- By default the ``run`` method takes exactly one argument: a data reference for the item of data to be processed.
+  Variations are possible, but require that you provide a :ref:`custom argument parser <creating-a-command-line-task-custom-argparse>` and often a :ref:`custom task runner <creating-a-command-line-task-custom-task-runner>`.
+- When run from the command line, most command-line tasks save the configuration used and the metadata generated.
+  See :ref:`creating-a-command-line-task-persisting-config-and-metadata` for more information.
+- They have an additional :ref:`class variable <creating-a-task-class-variables>`, ``RunnerClass``, that specifies a "task runner" for the task.
+  The task runner takes a parsed command and runs the task.
+  The default task runner will work for any script whose `run` method accepts a single data reference, such as ``ExampleCmdLineTask``.
+  If your task's `run` method needs something else then you will have to provide a :ref:`custom task runner <creating-a-command-line-task-custom-task-runner>`.
+- They have an additional :ref:`class variable <creating-a-task-class-variables>` ``canMultiprocess``, which defaults to `True`.
+  If your task runner cannot run your task with multiprocessing then set it `False`.
+  Note: multiprocessing only affects how the task runner calls the top-level task; thus it is ignored when a task is used as a subtask.
+
+.. _creating-a-command-line-task-run-script:
+
+Run Script
+==========
+
+A command-line task can be run as a pipeline via run script.
+This is usually a trivial script which merely calls the task's ``parseAndRun`` method. ``parseAndRun`` does the following:
+
+- Parses the command line, which includes determining the :ref:`configuration <creating-a-task-configuration>` for the task and which data items to process.
+- Constructs the task.
+- Calls the task's ``run`` method once for each data item to process.
+
+``examples/exampleCmdLineTask.py``, the runner script for ``ExampleCmdLineTask``, is typical:
+
+.. code-block:: python
+
+   #!/usr/bin/env python2
+   from lsst.pipe.tasks.exampleCmdLineTask import ExampleCmdLineTask
+   ExampleCmdLineTask.parseAndRun()
+
+For most command-line tasks you should put the run script into your package's :file:`bin/` directory, so that it is on your ``$PATH`` when you setup your package with eups.
+We did not want the run script for ``ExampleCmdLineTask`` to be quite so accessible, so we placed it in the :file:`examples/` directory instead of :file:`bin/`.
+
+Remember to make your run script executable using :command:`chmod +x`.
+
```

Comment:
Here's the place to remind readers that `parseAndRun()` will *not* ordinarily return control to the bin script (but see `--noExit` [or whatever I called that option])